### DISCUSSION TITLE
Compatibility with Assemblyscript

### DISCUSSION THREAD

**Discussion by fire** [statelyai/xstate#2761] [Ideas]
Any reports of compatibility with assemblyscript?

Wanted to try compiling to wasm.

**Comment by ChrisShank** [comment]
Havent heard of any attempts to do this. What do you think would be incompatible?

**Reply by fire** [reply]
Wanted to find evidence someone tried. Not sure where to start.

**Comment by drush** [comment]
I would be willing to support a bounty to add Assemblyscript support.  Alternatively, I'd be interested to setup a collaboration session with others who want to analyze the effort to enable Assemblyscript for `xstate-fsm`. Please indicate any interest in a bounty or participating on a pair session to work some of the basics out @fire @ChrisShank @edbek @Andarist

**Reply by Andarist** [reply]
I could discuss the idea - hashing out how it could be done and what kind of constraints such port would have etc.
